src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -183,7 +200,11 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
             var builder = new DefaultTagHelperContent();
 
-            if (mode == Mode.Fallback && string.IsNullOrEmpty(HrefInclude))
+            if (mode == Mode.FileVersion)
+            {
+                BuildLinkTag(attributes, builder);
+            }
+            else if (mode == Mode.Fallback && string.IsNullOrEmpty(HrefInclude))
```

Comment:
Think of the modes as primarily for detecting whether the helper will run or not. The modes overlap and combine when matched, so splitting the actual logic based on matched modes isn't ideal. Basically, what @NTaylorMullen said :smile: